frontend/src/services/fileService.ts

```diff
@@ -1,23 +1,57 @@
 import { request } from "./api";
 
 export async function selectFile(file: string): Promise<string> {
-  const data = await request(`/api/select-file?file=${file}`);
-  return data.code as string;
+  try {
+    const data = await request(`/api/select-file?file=${encodeURIComponent(file)}`);
+    if (typeof data.code !== 'string') {
+      throw new Error('Invalid response format: code is not a string');
+    }
+    return data.code;
+  } catch (error) {
+    console.error('Error selecting file:', error);
+    throw error;
+  }
 }
 
-export async function uploadFiles(files: FileList) {
-  const formData = new FormData();
-  for (let i = 0; i < files.length; i += 1) {
-    formData.append("files", files[i]);
-  }
+export async function uploadFiles(files: FileList): Promise<void> {
+  try {
+    const formData = new FormData();
+    Array.from(files).forEach(file => formData.append("files", file));
 
-  await request("/api/upload-files", {
-    method: "POST",
-    body: formData,
-  });
+    await request("/api/upload-files", {
+      method: "POST",
+      body: formData,
+    });
+  } catch (error) {
+    console.error('Error uploading files:', error);
+    throw error;
+  }
 }
 
 export async function listFiles(path: string = "/"): Promise<string[]> {
-  const data = await request(`/api/list-files?path=${path}`);
-  return data as string[];
+  try {
+    const data = await request(`/api/list-files?path=${encodeURIComponent(path)}`);
+    if (!Array.isArray(data)) {
+      throw new Error('Invalid response format: data is not an array');
+    }
+    return data;
+  } catch (error) {
+    console.error('Error listing files:', error);
+    throw error;
+  }
+}
+
+export async function saveFile(filePath: string, content: string): Promise<void> {
+  try {
```

Comment:
While we're at it: some safety checks could be added before try line, like
```ts
  if (!filePath || filePath.includes('..')) {
    toast.error('Invalid file path'); // needs translation too
  }
```

Reply:
I tried to fix it